valispace/__init__.py

```diff
@@ -110,23 +110,199 @@ def get_vali(self, id=None, name=None):
 				print("VALISPACE-ERROR: Vali id must be an integer")
 				return
 
+		if workspace_id:
+			try:
+				workspace_id = int(workspace_id)
+			except:
+				print("VALISPACE-ERROR: Workspace id must be an integer")
+				return
+			
+		if project_id:
+			try:
+				project_id = int(project_id)
+			except:
+				print("VALISPACE-ERROR: Project id must be an integer")
+				return
+
+		if parent_id:
+			try:
+				parent_id = int(parent_id)
+			except:
+				print("VALISPACE-ERROR: Parent id must be an integer")
+				return
+
+		if tag_id:
+			try:
+				tag_id = int(tag_id)
+			except:
+				print("VALISPACE-ERROR: Tag id must be an integer")
+				return
+
+		if valis_marked_as_impacted:
+			try:
+				valis_marked_as_impacted = int(valis_marked_as_impacted)
+			except:
+				print("VALISPACE-ERROR: Valis_marked_as_impacted must be an integer")
+				return
+
 		# Check if no argument was passed
-		if id is None and name is None:
-			print("VALISPACE-ERROR: 1 argument expected (name or id)")
+		if id is None and name is None and workspace_id is None and project_id is None and parent_id is None and parent_name is None and tag_id is None and tag_name is None and valis_marked_as_impacted is None:
+			print("VALISPACE-ERROR: at least 1 argument expected")
 			return
 
-		# Check if name argument was passed
-		if id is None:
-			id = self.__name_to_id(name)
+		# Increment function to add multiple fields to url
+		def increment(url):
+			if not url.endswith('?'):
+					url += "&"
+			return url
+
+		# URL
+		url = self.valispace_login['url'] + "vali/?"
+		if id:
+			url += "id=" + str(id)
+		elif name:
+			url += "shortname=" + str(name)
+		else:
+			url += "?"
+			if workspace_id:
+				url += "parent__project__workspace=" + str(project_id)
+			if project_id:
+				url = increment(url) + "parent__project__id=" + str(project_id)
+			if parent_id:
+				url = increment(url) + "parent__id=" + str(parent_id)
+			if parent_name:
+				url = increment(url) + "parent__name=" + str(parent_name)
+			if tag_id:
+				url = increment(url) + "tags__id=" + str(tag_id)
+			if tag_name:
+				url = increment(url) + "tags__name=" + str(tag_name)
+			if valis_marked_as_impacted:
+				url = increment(url) + "valis_marked_as_impacted=" + str(valis_marked_as_impacted)
 
-		# Access API
-		url = self.valispace_login['url'] + "vali/" + str(id) + "/"
 		headers = self.valispace_login['options']['Headers']
+		response = requests.get(url, headers=headers)
+
+		return response.json()
+
+	def get_component(self, id=None, name=None, workspace_id=None, project_id=None, project_name=None, parent_id=None, tag_id=None, tag_name=None):
```

Comment:
Same comments as for the vali: perhaps better to split into get and filter function, it would be good to have also the workspace and project name, and in the comment it needs to be explained what are the input and output formats.